reframe/core/pipeline.py

```diff
@@ -1124,6 +1124,24 @@ def check_performance(self):
             return
 
         with os_ext.change_dir(self._stagedir):
+
+            # Check if default reference perf values are provided and
+            # store all the variables  tested in the performance check
+            is_default_present = False
+            variables = set()
+            for system, perf_data in self.reference.items():
+                variables.add((system.split(":")[-1], perf_data[-1]))
```

Comment:
You should handle also the case that there are no units defined.

Reply:
Is it possible to provide a reference value without unit?

Reply:
Yes the unit is optional. Am I right @vkarak ?

Reply:
I may be wrong but I remember a discussion about rejecting performance tests without units in order to avoid mistakes